src/main/java/com/minecolonies/core/colony/managers/TravellingManager.java

```diff
@@ -0,0 +1,213 @@
+package com.minecolonies.core.colony.managers;
+
+import com.google.common.collect.Maps;
+import com.minecolonies.api.colony.ICitizenData;
+import com.minecolonies.api.colony.IColony;
+import com.minecolonies.api.colony.managers.interfaces.ITravellingManager;
+import com.minecolonies.api.entity.citizen.AbstractEntityCitizen;
+import com.minecolonies.api.util.EntityUtils;
+import com.minecolonies.api.util.Log;
+import com.minecolonies.api.util.constant.ColonyConstants;
+import com.minecolonies.api.util.constant.NbtTagConstants;
+import com.minecolonies.core.util.TeleportHelper;
+import net.minecraft.core.BlockPos;
+import net.minecraft.nbt.CompoundTag;
+import net.minecraft.nbt.ListTag;
+import net.minecraft.nbt.NbtUtils;
+import net.minecraft.nbt.Tag;
+import net.minecraftforge.common.util.INBTSerializable;
+
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+
+public class TravellingManager implements ITravellingManager, INBTSerializable<CompoundTag>
+{
+
+    private final IColony                    colony;
+    private final Map<Integer, TravelerData> travelerDataMap = Maps.newHashMap();
+
+    public TravellingManager(final IColony colony) {this.colony = colony;}
+
+    @Override
+    public boolean isTravelling(final int citizenId)
+    {
+        return Optional.ofNullable(travelerDataMap.get(citizenId)).map(TravelerData::isTraveling).orElse(false);
```

Comment:
That seems like a weird unpacking chain that could be done more simple without packing it in an optional first

Reply:
still

Reply:
I would have it just spelled out without the optional wrapping tbh.

Reply:
Does it matter? Because we just had a discussion on formatting..... And you agreed that this is nitpicking to far.

Reply:
Individually it doesn't matter. It makes it a harder to debug if there is a problem in there and a bit harder to read compared to written out. And, if we do this unpacking in a lot of places it creates more for the GC to handle.